<launch>

     <!-- 启动运控 -->
    <node pkg="fastlio" type="g1_ctrl_cwk.py" name="g1_ctrl_cwk" cwd="node" output="screen"/>

    <node pkg="tf" type="static_transform_publisher" name="base_to_body" args="0.0 0 0.0 3.1416 3.1416 0 /body /base_link 1" /> 
    <include file="$(find livox_ros_driver2)/launch_ROS1/msg_MID360.launch">
    </include>
    <rosparam command="load" file="$(find fastlio)/config/localize.yaml" />
    <node pkg="fastlio" type="localizer_node" name="localizer_node" output="screen"/>


    <arg name="pcd_path" default="platform_1_2.pcd" />
    <arg name="pcd_dir" default="/home/unitree/HongTu/G1Nav2D/src/fastlio2/PCD" />

    <node pkg="fastlio" type="slam_reloc.py" name="slam_reloc" output="screen">
        <param name="pcd_path" value="$(arg pcd_dir)/$(arg pcd_path)" />
    </node>


    <arg name="map_file" default="platform_1_2.yaml" />

    <include file="$(find fastlio)/launch/gridmap_load.launch">
            <arg name="rviz" value="false"/>
            <arg name="2dmap_file" value="$(arg map_file)" />


    </include>

    <arg name="ground_map_path" default="ground_platform_1_2.pcd" />
    <arg name="ground_map_dir" default="/home/unitree/HongTu/G1Nav2D/src/fastlio2/PCD" />

    <include file="$(find tool)/launch/downsample_pointcloud.launch">
        <arg name="pcd_file" value="$(arg ground_map_dir)/$(arg ground_map_path)" />

    </include>

    <include file="$(find xju_pnc)/launch/move_base.launch" />

    <include file="$(find pointcloud_to_laserscan)/launch/point_to_scan.launch" />

    <include file="$(find tool)/launch/body2any_pointcloud.launch">
    </include>


    <include file="$(find velocity_smoother_ema)/launch/velocity_smoother_ema.launch">
    </include>

    <arg name="use_rviz" default="true" />

    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find fastlio)/rviz/localize.rviz" if="$(arg use_rviz)" />


    <!-- <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find fastlio)/rviz/localize.rviz" /> -->
</launch>
